react/components/ProductImages/components/Carousel/index.js

```diff
@@ -144,10 +140,11 @@ class Carousel extends Component {
   }
 
   componentDidUpdate(prevProps) {
-    const { gallerySwiper, loaded, activeIndex } = this.state
-    const { isVideo } = this
+    const { loaded, activeIndex } = this.state
+    const isVideo = this.isVideo
+    const gallerySwiper = path(['swiper'], this.gallerySwiper.current)
 
-    if (prevProps.slides !== this.props.slides) {
+    if (!equals(prevProps.slides, this.props.slides)) {
```

Comment:
What condition does `!equals` verifies that `!==` is not enough?

Reply:
`!==` evaluates to true if the elements have different *references*, while `!equals` compares the values in the objects inside the array.